cpp/src/plasma/client.cc

```diff
@@ -144,6 +165,11 @@ uint8_t* PlasmaClient::lookup_mmapped_file(int store_fd_val) {
   return entry->second.pointer;
 }
 
+bool PlasmaClient::IsInUse(const ObjectID& object_id) {
```

Comment:
This should be exposed through the Info method, see https://github.com/apache/arrow/blob/ecb7605a640dc34452dd696240a74df9a9f7f39b/cpp/src/plasma/client.h#L301

Given that is is private, we can leave it for now, but can you create a JIRA for it?

Reply:
Ok, see ARROW-2379

Reply:
`Info` requires an IPC to the plasma store and seems too heavy-weight since this can all be determined with locally available information.

`IsInUse` is probably better named something like `IsInUseByThisClient` (that's an ugly name, but I think it's more clear).

We have a lot of code in `client.cc` that's already doing stuff like

```cpp
auto elem = objects_in_use_.find(object_id);
if (elem == objects_in_use_.end()) {
  ...
}
```

so it may make sense for `IsInUse` to replace some of that code.